WordPress/gutenberg
# Nav Block - show recent pages as default suggestions when creating Nav Links

<img width="535" alt="Screenshot 2020-01-08 at 11 37 27" src="https://user-images.githubusercontent.com/444434/71975222-4745b080-320b-11ea-922a-3627ee448041.png">

## Description
Closes https://github.com/WordPress/gutenberg/issues/18899.

When first creating a Nav Link ("item") Block the hyperlink UI presents a blank search field. To better serve the default use case of adding pages, this PR updates the implementation to display a list of the most recently created Pages.

This involves a refactor of the `URLInput` to add a new boolean prop `initialSuggestions` which (internally) ~affords the ability to trigger rendering of search results without anything being entered into the `<input>` field~ (for more on this [see the original Issue](https://github.com/WordPress/gutenberg/issues/18899#issuecomment-571520975)) triggers a call for default search results when the input field is empty.

## How has this been tested?
1. Add Nav Block
2. Insert new Nav Link Block.
3. See a list of most recently created Pages displayed as the default search results in the hyperlink creation UI.

## Screencapture 

![Screen Capture on 2020-01-08 at 11-35-37](https://user-images.githubusercontent.com/444434/71975123-182f3f00-320b-11ea-8fc0-75f19aa111fe.gif)


## Types of changes
New feature (non-breaking change which adds functionality).

## Checklist:
